### PR TITLE
cleanup, remove last uses of deprecated tribe.carrier2 and tribe.carrier

### PR DESCRIPTION
**Pull request by SimonHeimberg**
**Type of change**
Cleanup

**Issue(s) closed**
Fixes -

### CODE REVIEW

File: test/maps/lua_testsuite.wmf/scripting/tribes_descriptions.lua [hunk 1]
```diff
@@ -51,13 +51,13 @@ function test_descr:test_get_builder()
    assert_equal("atlanteans_builder", tribe.builder)
 end
 
-function test_descr:test_get_carrier()
+function test_descr:test_get_carriers()
    local tribe = egbase:get_tribe_description("atlanteans")
+   assert_equal("atlanteans_carrier", tribe.carriers[1])
+   assert_equal("atlanteans_horse", tribe.carriers[2])
+   assert_equal(2, #tribe.carriers)
+   -- deprecated, test until removed:
```
Comment (Noordfrees) [inline test/maps/lua_testsuite.wmf/scripting/tribes_descriptions.lua]: What we mean by "deprecated" in the documentation is that `carriers` is now preferred but `carrier` and `carrier2` will continue to work for API/add-ons compatibility. No removal of deprecated functionality is planned unless absolutely unavoidable.
```suggestion
   -- Deprecated, but shall still work for API compatibility:
```

File: data/scripting/infrastructure.lua [hunk 2]
```diff
@@ -65,8 +65,8 @@ function connected_road(roadtype, p, start, cmd, g_create_carriers)
          r = p:place_road(roadtype, start, table.unpack(moves))
          start = r.end_flag
          if create_carriers then
-            if roadtype == "normal" then r:set_workers(p.tribe.carrier, 1)
-            elseif roadtype == "busy" then r:set_workers({[p.tribe.carrier] = 1, [p.tribe.carrier2] = 1})
+            if roadtype == "normal" then r:set_workers(p.tribe.carriers[1], 1)
+            elseif roadtype == "busy" then r:set_workers({[p.tribe.carriers[1]] = 1, [p.tribe.carriers[2]] = 1})
```
Comment (Noordfrees) [inline data/scripting/infrastructure.lua]: Actually I wonder if what is desired here is not to place *one of each* type of carrier on the busy road. This is only really relevant for scripts for the Europeans add-on currently but this probably shouldn't be hard to fix while you're already modernizing this line.

**Review by Noordfrees — COMMENTED** [review]
LGTM. One optional suggestion inline.

File: data/scripting/infrastructure.lua (hunk 2 above)
Comment (hessenfarmer) [reply data/scripting/infrastructure.lua]: @SimonHeimberg the 2 lines could be made one line using the valid_workers property of the road.

Comment (SimonHeimberg) [reply data/scripting/infrastructure.lua]: Done. Almost at least. Does the ferry need different handling?

Comment (hessenfarmer) [reply data/scripting/infrastructure.lua]: @Noordfrees looks good to me valid_workers should include ferries.

**Review by Noordfrees — APPROVED** [review]

**Pull request merged**